content/js1/sprints/1/prep/index.md

```diff
@@ -8,4 +8,443 @@ backlog= 'Module-JS1'
+
+If we type `Math.round(10.3)` then again we get the result 10.
+So we say that `Math.round(10.3)` outputs 10.
+
+{{<note title="Note" type="note">}}
+It's important to note that `Math.round` is **implementation opaque**.
+**Implementation opaque** means we can’t read the set of instructions for `Math.round`. This is because `Math.round` was authored by the developers that created NodeJS.
+{{</note>}}
+
+### 📁 Running scripts
+
+So far we’ve seen how expressions can be evaluated using the Node REPL. The Node REPL is a very useful tool for evaluating expressions quickly.
+
+However, most of the time, we need to write more complex programs in files that consist of many lines.
+Instead of using the REPL mode, we can get Node to execute the instructions written in our file.
+
+We can use the node command to run a JavaScript file.
+A JavaScript file ends with `.js` - this is known as the file extension.
+
+Let’s suppose we have a file `sums.js` like this:
+
+```js title="sums.js"
+10 + 3;
+10 * 3;
+10 / 3;
+```
+
+In this case, we can write the command `node sums.js`.
+
+This command is an instruction to execute the program written inside index.js. Our program consists of 3 lines, each line an expression. So the machine will start:
+
+`10 + 3;` the computer will evaluate this expression  
+`10 * 3;` the computer will evaluate this expression  
+`10 / 3;` the computer will evaluate this expression
+
+{{<note title="Activity" type="activity">}}
+
+Try out the following steps below to check you can run a file with Node:
+
+1. In your terminal, create a new file called `example.js`.
+
+2. Try writing a few expressions in the file like the ones above.
+
+3. Get Node to run this file. ( Don't use the REPL now - there should be a command to execute the whole file.)
+
+{{</note>}}
+
+Once the computer evaluates these expressions the execution of the program is complete. But, we’re left with a problem. With the REPL, the user inputs an expression, the computer evaluates it and then prints the result. It then loops back and prompts the user for another input. But now the computer will go and execute each line sequentially until completion.
+
+So this new problem can be expressed as a question:
+
+> ❓ Problem
+>
+> "How can we check the values of variables in our program when it is being executed?"
+
+### 🖨️ Printing to the terminal
+
+In order to check the values when our program runs, we can make use of a function called `console.log`.  
+`console.log` will allow us to log expressions to the console while our program is executing. This is very useful for complex programs when we need to check the value of expressions in our program.
+
+Let's see how to use `console.log`
+
+In a file called `example.js`, we can write name of the function `console.log`.
+
+```js title="example.js"
+console.log;
+```
+
+However, if we run this file with Node, we won't be able to see anything in the terminal.
```

Comment:
I think this block is probably confusing. I'd maybe go the other way around; something like:

````
Just like when we called `Math.round` by adding parentheses with an input value inside them, we call `console.log` in the same way:

```js title="example.js"
console.log("hello there!");
```

Try running the file containing this instruction. You should see the string "hello there!" logged in the terminal.

If you go back to a REPL, and evaluate `console.log`, you'll see it's a function, just like we saw with `Math.round`:

[screenshot]

If we just wrote `console.log` in our file and ran the file, we wouldn't expect to see anything because we're not _calling_ the function. We know that the expression `console.log` evaluates to a function, but until we call it, it doesn't _do_ anything."
````

I'd be tempted to _also_ move the "Logging vs function output" (i.e. side effect vs return value) explanation here, and then use the "Saving function output" section as a "Let's reinforce that" section.